lang/java/reef-runtime-yarn/src/main/java/org/apache/reef/runtime/yarn/client/YarnJobSubmissionHandler.java

```diff
@@ -147,4 +144,14 @@ private String getQueue(final JobSubmissionEvent jobSubmissionEvent,
                           final String defaultQueue) {
     return jobSubmissionEvent.getQueue().orElse(defaultQueue);
   }
+
+  private String getUserBoundJobSubmissionDirectory(Configuration configuration) {
+    try {
+      return Tang.Factory.getTang().newInjector(configuration).getNamedInstance(DriverJobSubmissionDirectory.class);
+    } catch (InjectionException ex) {
+      return null;
```

Comment:
You could use `Optional` here and change the ternary operator above to a call to `orElse`.